### DISCUSSION TITLE
Github artifacts improvements

### DISCUSSION THREAD

**Discussion by wpotrzebowski** [SasView/sasview#2239] [Code Camp Topics]
Items to discuss
- Easy access to github actions artifacts (also for external users)
- Testing installers as a part of GH actions

**Comment by llimeht** [comment]
> Testing installers as a part of GH actions

I had a first play with what this might look like the other day. I'll get something into a shareable form so that we have a chance to see what is possible and estimate the work required. As I mentioned the other day, this is likely a piece of work that would make a good internship/summer of code project.

**Comment by lucas-wilkins** [comment]
> Easy access to github actions artifacts (also for external users)

We should probably get on top of minor versions

> Testing installers as a part of GH actions

Yes!

**Comment by llimeht** [comment]
For testing out the installer, here's a draft of some updated CI that:

- removes the repetition from the yml files
- simplifies the yml, updating to some newer ways of doing things
- tests that the installer can actually be installed (not on macos yet)
- tests that the installed version runs (not on macos yet)

This approach was able to spot the recent problem we were having with `setuptools._distutils` and also verify that it had been fixed.

Work-in-progress tree: https://github.com/llimeht/sasview/compare/main...llimeht:sasview:tmp/cicombinematrix

Example CI run output: https://github.com/llimeht/sasview/actions/runs/3257679776

There's the start of some work to try to take screenshots of SasView while running on CI. The intention was that the CI run would be able to perform some UI tests to give confidence in integration, functionality, lazy loading of modules etc. Screenshots would also give a visual record from which potential UI regressions could be tracked. The screenshotting isn't quite working as planned at this stage; not really sure why. I suspect we should land something that tests the installer first, and come back to screenshotting later.
